base_types.go

```diff
@@ -0,0 +1,29 @@
+// This package contains various type-related base classes intended
+// to be used in composition across type structures in this project.
+
+package linodego
+
+// baseType is a base struct containing the core fields of a resource type
+// returned from the Linode API.
+type baseType[PriceType any, RegionPriceType any] struct {
```

Comment:
I decided to use a generic base struct so we can still share common type fields while allowing resource types of diverge without a breaking change if necessary.